File: Week 5/X38951_en/sliding_window.py
'''
SLINDING-WINDOW(S, x, y)
let A be an empty array
n <- |S|
i <- 0
while i <= n-x do
    append S[i,...,i+x] to A
    i <- i + y
return A
'''


import sys


def sliding_window(s, x, y):
    A = []
    n = len(s)
    i = 0
    while i <= n - x:
        A.append(s[i:i + x])
        i += y
    return A


inp = sys.stdin.read().splitlines()
for e in sliding_window(inp[0], int(inp[1]), int(inp[2])):
    print(e)

File: Week 5/X38951_en/sliding_window.cpp
/*
SLINDING-WINDOW(S, x, y)
let A be an empty array
n <- |S|
i <- 0
while i <= n-x do
    append S[i,...,i+x] to A
    i <- i + y
return A
*/

#include <iostream>
#include <string>
#include <vector>
using namespace std;

vector<string> sliding_window(string s, int x, int y) {
    vector<string> A;
    int n = s.length();
    int i = 0;
    while (i < n - x + 1) {
        A.push_back(s.substr(i, x));
        i = i + y;
    }
    return A;
}

int main() {
    string s;
    int x;
    int y;
    cin >> s >> x >> y;
    vector<string> res = sliding_window(s, x, y);
    for (auto i = res.begin(); i != res.end(); ++i)
        cout << *i << endl;
    return 0;
}
